docs/designers-developers/developers/tutorials/javascript/scope-your-code.md

```diff
@@ -0,0 +1,92 @@
+# Scope your code
+
+Historically, JavaScript files loaded by the same document share the same scope. This means that a global variable declared in a file will be seen by the code in other files.
+
+This is `one.js` file:
+
+```js
+var yourPluginName = 'MyPlugin';
+console.log( 'Plugin name is ', yourPluginName );
+```
+
+This is `two.js` file:
+
+```js
+var yourPluginName = 'YourPlugin';
+var yourName = 'A great JS developer!';
+console.log( 'Plugin ', yourPluginName, ' authored by ', yourName );
+```
+
+If you load them both in a page, they'll overwrite each other global variables.
+
+**First step: scope the code within a function**
+
+In order to fix this, you need to scope your code. In JavaScript, you do this by writing it within a function, because they create a _local scope_. And, to prevent polluting the global scope with your function's name, you should write your code within an _anonymous_ function.
+
+Scoped `one.js` code:
+
+```js
+function() {
+	var yourPluginName = 'my plugin';
+	console.log( 'Plugin name is ', yourPluginName );
+}
+```
+
+And `two.js`:
+
+```js
+function() {
+	var yourName = 'A great JS developer!';
+	console.log( 'Plugin ', yourPluginName, ' authored by ', yourName );
+}
+```
+
+With this trick, the different files won't override each other. They also won't work as intended! Nothing will be displayed in the console because these functions are being called by no one.
```

Comment:
```suggestion
With this trick, the different files won't override each other's variables. Unfortunately, they also won't work, because these functions are being called by no one. We've only _defined_ the functions; we haven't _executed_ them yet.
```